app/components/Receive/QRCodeForm/index.jsx

```diff
@@ -59,17 +65,27 @@ export default class QRCodeForm extends React.Component<Props, State> {
               <div className={styles.inputDescription}>ASSET</div>
               <AssetInput
                 symbols={symbols}
-                value={asset}
-                onChange={value => this.setState({ asset: value })}
+                value={{ label: asset, value: asset }}
+                onChange={value => this.setState({ asset: value, amount: 0 })}
               />
             </div>
             <div className={styles.amount}>
               <div className={styles.inputDescription}>AMOUNT</div>
               <NumberInput
                 value={amount}
                 placeholder="Amount"
-                options={{ numeralDecimalScale: COIN_DECIMAL_LENGTH }}
-                onChange={value => this.setState({ amount: value })}
+                options={{
+                  numeralDecimalScale: 8
+                }}
+                // this is a hack because Cleave will not update
+                // when props change https://github.com/nosir/cleave.js/issues/352
```

Comment:
Was spinning my wheels on this for a few hours decided to just move on with this implementation... Updating any props other than value does nothing super frustrating